content/tutorials/single-page-tutorials/mint-nft-with-javascript-shimmer.md

```diff
@@ -0,0 +1,239 @@
+---
+title: How to mint a native NFT with Javascript on Shimmer
+description: Learn how to mint an NFT on the Shimmer Network with Javascript
+tags: [Shimmer, NFT, beginner, Javascript, iota.js, client]
+---
+
+# How to Mint a Native NFT With Javascript on Shimmer
+
+A non-fungible token (NFT) is a token that is verifiably unique and can be used to prove and transfer ownership of a digital asset. You can create NFTs in the Shimmer ledger as a native asset or use smart contracts as a smart asset. This tutorial uses Javascript to mint a native NFT in the Shimmer ledger.
+
+## What you will learn
+
+- [How NFTs work on Layer 1 (L1)](#What_is_an_NFT?)
+- [How you can mint an NFT](#Mint_a_NFT)
+- [How to use and read the Shimmer Explorer](#)
+- [How the Immutable Metadata works](#How_to_add_Metadata_to_an_NFT)
+
+:::info Why we have iota.js and iota.rs with Node.js bindings
+- You can use [iota.js](https://wiki.iota.org/iotajs/welcome) in a browser. This is not possible with the Node.js bindings on iota.rs.
+- [iora.rs Node.js](https://wiki.iota.org/iota.rs/getting_started/nodejs) bindings are recommended to use in Node.js applications.
+- For wallet integrations, we recommend using [wallet.rs](https://wiki.iota.org/wallet.rs/welcome), which offers abstractions to handle IOTA payments and can optionally interact with IOTA Stronghold for seed.
+:::
+
+# Prerequisites
+- [Node.js](https://nodejs.org/en/)
+
+## What is an NFT?
+
+An NFT is short-term for Non Fungible Tokens - so it's a unique token that can represent a digital asset within its immutable data. This is a new decentral way to prove ownership and brings a lot of functionality and expandability which can be used in DApp and Games. An NFT can represent tangible assets like a house or a car or digital assets like a membership. The common use case is NFTs with linked Images, which often bring special membership, and access to private places.
+
+Here in the tutorial, we will mint an NFT with a permanently linked image. The term `mint` means the process of creating a new NFT in the network. 
+
+
+## Set Up 
+Let's start with preparing the development environment. This tutorial assumes that you have Node.js installed.
+
+First, copy the iota.js library to your computer, so you can run the example and go through it. 
+
+```bash
+git clone --branch feat/stardust https://github.com/iotaledger/iota.js shimmer-nft-example
+```
+
+Now you should switch to the example directory in the `mint-nft`example.
+
+```bash
+cd shimmer-nft-example/packages/iota/examples/mint-nft
+```
+
+You will need to update the iota.js library for the Stardust update and install all the dependencies of the example.
+
+```bash
+npm install @iota/crypto.js@next @iota/iota.js@next @iota/util.js@next @iota/pow-neon.js @types/node --save-dev
```

Comment:
Can't run this command.
I get: `npm ERR! Invalid Version: `
What am I doing wrong?

Reply:
From the docs: https://wiki.iota.org/iotajs/how_tos/run_how_tos you need to do

```
npm install
npm run dist-neon
npm run dist-all
```

to be able to run their examples.

Reply:
The idea here was to install the deps by the latest npmjs modules (@next), so you don't need to build every package - because you also would need a rust compiler to run this javascript tutorial. But it seems, this is nor working on many devices. 

What do you think? It's not a good DX to enforce javascript developers install a rust compiler. Are they other ways to solve that?

Reply:
I guess at this point I'd actually follow the format that for example @adrian-grassl uses in his tutorials: tell the reader how to set up a fresh project with its own dependencies and use the example code given by the iotajs repo only as a reference. This removes the need for the reader to clone the entire repo and navigate to an obscure directory altogether and to me having to install and alter dependencies in an example folder myself feels weird anyway. Just tell the reader what is the next step and give the code to do that so the reader can copy it. Wdyt?

Reply:
A "fresh-project-start" for a tutorial sounds like a very good practice. The idea was here, to reduce to copy paste amount for the developers, because this examples has 300 lines of code - which is too much for a "step by step" tutorial in my opinion. But this example acts more like a script which leads you trough the process in your terminal, maybe this is just too heavy for a tutorial and we can reduce the code. 

But first, we need to check the `npm install` part, im not sure if this removes the error with the dependency installation.